### PR TITLE
How to set catalog prefer as public in vscode setting?

### PR DESCRIPTION
**Pull request by angelozerr**
How to set catalog prefer as public in vscode setting?

See https://github.com/redhat-developer/vscode-xml/issues/295

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/settings/ContentModelSettings.java [hunk 1]
```diff
@@ -64,7 +67,13 @@ public void setCatalogs(String[] catalogs) {
...
 	 */
 	public String[] getCatalogs() {
-		return catalogs == null ? new String[0] : catalogs;
+		XMLCatalogSettings catalog = getCatalog();
+		if (catalog != null) {
+			if (catalog.getFiles() != null) {
+				return catalog.getFiles();
+			}
+		}
+		return catalogs;
```
Comment (fbricon) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/settings/ContentModelSettings.java:76]: it used to never return null, you should keep that

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/settings/XMLCatalogSettings.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ */
+public class XMLCatalogSettings {
+
+	public static Prefer DEFAULT_PREFER_PUBLIC = Prefer.PUBLIC;
+
+	public static boolean DEFAULT_USE_PREFER_PUBLIC = true;
+
+	public static boolean DEFAULT_USE_LITERAL_SYSTEM_ID = true;
+
+	public static final XMLCatalogSettings DEFAULT_CATALOG = new XMLCatalogSettings();
```
Comment (fbricon) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/settings/XMLCatalogSettings.java:26]: default catalog should be immutable

**Comment by fbricon** [comment]
please use a proper commit message (explaining the change)

**Comment by angelozerr** [comment]
I close this issue since original issue https://github.com/redhat-developer/vscode-xml/issues/295 doesn't require this new settings.

**Pull request closed without merging**